### ISSUE TITLE
Autocommand error when setting filetype

### ISSUE THREAD

**Issue by clason** [neovim/nvim-lspconfig#1524]
### Description

When setting the filetype of an empty buffer to one for which a server is configured to attach, an error is thrown:
```
Error detected while processing FileType Autocommands for "lua":
E5108: Error executing lua /usr/local/share/nvim/runtime/lua/vim/uri.lua:62: attempt to index local 'path' (a nil value)
stack traceback:
        /usr/local/share/nvim/runtime/lua/vim/uri.lua:62: in function 'uri_from_fname'
        ...ck/packer/start/nvim-lspconfig/lua/lspconfig/configs.lua:191: in function '_make_config'
        .../pack/packer/start/nvim-lspconfig/lua/lspconfig/util.lua:255: in function 'add'
        ...ck/packer/start/nvim-lspconfig/lua/lspconfig/configs.lua:220: in function 'try_add'
        [string ":lua"]:1: in main chunk
```

### Neovim version

NVIM v0.7.0-dev+661-g99f6260f8
Build type: Release
LuaJIT 2.1.0-beta3

### Nvim-lspconfig version

5b8624c1bcd332e9fd7ae33a2ca910adb31a7ae7

### Operating system and version

macOS 12.0

### Affected language servers

all

### Steps to reproduce

1. `nvim -u minimal_init.lua`
2. `set ft=python`

(with the stock provided `minimal_init.lua`, which sets up `pyright`)

### Actual behavior

error (see above)

### Minimal config

```Lua
curl -fLO https://raw.githubusercontent.com/neovim/nvim-lspconfig/master/test/minimal_init.lua
```

### Health check

lspconfig: require("lspconfig.health").check()
========================================================================
  - INFO: pyright: configuration checked.

### LSP log

N/A

**Comment by d9d6ka** [comment]
The problem is also exists in neovim v0.6.0.
If I open the Python-file via `:e` command then no problem occurs.

But when I try to change Filetype then I receive the same error as @clason 

If I try to open the file via `dirvish` I receive the similar error. `add` and `try_add` functions are there as well.
```
Обнаружена ошибка при обработке FileType Автокоманды для "python":
E5108: Error executing lua ....0-dev-665-g4e352ecc4\share\nvim\runtime/lua/vim/lsp.lua:239: cmd_cwd: expected directory, got D:/git/advent-of-code/2015/day-07/part1.py
stack traceback:
        [C]: in function 'error'
        vim/shared.lua:608: in function 'validate'
        ....0-dev-665-g4e352ecc4\share\nvim\runtime/lua/vim/lsp.lua:239: in function 'validate_client_config'
        ....0-dev-665-g4e352ecc4\share\nvim\runtime/lua/vim/lsp.lua:688: in function 'start_client'
        ...\pack\packer\start\nvim-lspconfig/lua/lspconfig/util.lua:267: in function 'add'
        ...ck\packer\start\nvim-lspconfig/lua/lspconfig/configs.lua:220: in function 'try_add'
        [string ":lua"]:1: in main chunk
E5108: Error executing lua ....0-dev-665-g4e352ecc4\share\nvim\runtime/lua/vim/lsp.lua:239: cmd_cwd: expected directory, got D:/git/advent-of-code/2015/day-07/part1.py
stack traceback:
        [C]: in function 'error'
        vim/shared.lua:608: in function 'validate'
        ....0-dev-665-g4e352ecc4\share\nvim\runtime/lua/vim/lsp.lua:239: in function 'validate_client_config'
        ....0-dev-665-g4e352ecc4\share\nvim\runtime/lua/vim/lsp.lua:688: in function 'start_client'
        ...\pack\packer\start\nvim-lspconfig/lua/lspconfig/util.lua:267: in function 'add'
        ...ck\packer\start\nvim-lspconfig/lua/lspconfig/configs.lua:220: in function 'try_add'
        [string ":lua"]:1: in main chunk
```

**Comment by clason** [comment]
It's a bug in `nvim-lspconfig`, not in neovim (hence the issue here), so this is not at all surprising. The cause is known and a WIP fix is #1525

**Comment by d9d6ka** [comment]
I've updated all the packages and the problem is still there...
I'm receiving the same errors I've posted above when I try to open any file not via command line or `:e` command.
